### PR TITLE
Sử dụng gem devise

### PR DESCRIPTION
**Pull request by nguyenxuantuan**
https://edu-redmine.framgia.vn/issues/4419

### CODE REVIEW

**Comment by nguyenxuantuan** [comment]
ready

File: .gitignore [hunk 1]
```diff
@@ -1,50 +1,35 @@
...
+.byebug_history
 
-# unless supporting rvm < 1.11.0 or doing something fancy, ignore this:
-.rvmrc
+# Ignore master key for decrypting credentials and more.
+/config/master.key
+/test/*
+/config/database.yml
+/config/secrets.yml
+/log/development.log
```
Comment (huongnguyenmta) [inline .gitignore]: end file

File: config/credentials.yml.enc [hunk 2]
```diff
@@ -0,0 +1 @@
+31gLRyNKMAczkU/Eov8WUpDi3DvM2i71c8TOkYZcnRUE5Zf4y02vEuog1xuM9cPCmXsfeuaZl9wiqmpE7honKq7oiMga7LMMNYC9n1ISU8ygjtW/25BvvG0ymjxa4R1nue/l8XPgvccEuNR2kdvl1yKWW9q2TNp8srQ6SG1HC7wFOhsd/lwpfuPOIimax/rK2dJcH/rmq/H44eR22XcVI6hziqKXYvgLubobrXPdYvaPYvqOcXjMgiIJzWw5EjaS1m2fdFEaCeTwH+vgdSU0N+AzlHEBq1cPp4FtyqebxcHdeP14kfW8gWMLWbmQQTvVbyJZAnXBzsRcv9U/XUUOzWMUXJSdSaN6d/68ot13uIFLorqCXA/iNx+Qo/CT0nWzURXjhjf1bOh6zVSLl3fMniyEQeJArCV5Rihq--0PwXF9lPh7273oS3--OuhWDQWmZxMASMrY5gr/ng==
```
Comment (huongnguyenmta) [inline config/credentials.yml.enc]: move this file into `.gitignore`

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class ApplicationController < ActionController::Base
+	before_action :set_locale
+	
+  def set_locale
+		locale = params[:locale].to_s.strip.to_sym
```
Comment (huongnguyenmta) [inline app/controllers/application_controller.rb]: indent

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<html>
+  <head>
+    <title>PR2Oe12TestManagement</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+    <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track": "reload" %>
```
Comment (huongnguyenmta) [inline app/views/layouts/application.html.erb]: redundant space

**Comment by huongnguyenmta** [comment]
commented
- update information in this ticket redmine
- one more comments

**Pull request closed without merging**